File: backend_laravel_starter/app/Models/Workout.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Workout extends Model
{
    protected $fillable = [
        'title',
        'slug',
        'description',
        'difficulty',
        'duration_minutes',
        'is_active',
    ];

    protected $casts = [
        'is_active' => 'boolean',
    ];

    public function exercises()
    {
        return $this->belongsToMany(Exercise::class)
            ->withPivot(['position', 'sets', 'reps', 'rest_seconds'])
            ->orderBy('workout_exercises.position');
    }
}

File: lib/data/models/workout.dart
import 'exercise.dart';

class Workout {
  final String id;
  final String title;
  final String subtitle;
  final int durationMinutes;
  final List<Exercise> exercises;

  const Workout({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.durationMinutes,
    required this.exercises,
  });
}
